### PR TITLE
Improve cli config options and types for config

### LINKED ISSUE #10: Add option for including specified modules/path in webpack config 'include'
Required when we need to import some node-modules' file.

### PR DESCRIPTION
**Pull request by rohit-gohri**
- Read config in this order:
  - `sm-webpack.js`; if not found then
  - `sm-webpack` key `sm-config.js`; if not found then
  - `sm-webpack` key in `package.json`
- Also now pass `webpackConfig` field as `webpackConfig` to merge
- Add index.d.ts file with types for config
- Fix `getProdConfig` and `getDevConfig` functions
- Disable cache-loader for babel if debug is true
- Close #10

### CODE REVIEW

**Comment by hit12369** [comment]
Problem with `package.json` is that it's `json` and not `js` so very restrictive (regards to comments and all). While it's okay to read it, we should also support a common `js` file. (Maybe `config.js` or `sm-config.js`, we can directly use `cfg` from `sm-utils` too).

**Comment by rohit-gohri** [comment]
@hit12369 Review needed

File: index.js [hunk 1]
```diff
@@ -31,9 +31,7 @@ function runWebpack({env, config, webpackConfig}) {
 			if (!config.isSSR) {
 				console.log(formatStats(stats, config.destPath));
 			}
-
-			// give time to webpack bundle analyzer to output stats
-			setImmediate(resolve);
+			resolve();
```
Comment (hit12369) [inline index.js:34]: I think this might create problems, `setImmediate(resolve);` is fine or that creates any issues?

Comment (rohit-gohri) [reply index.js:34]: This was no longer needed as it was fixed in bundle analyzer itself: https://github.com/webpack-contrib/webpack-bundle-analyzer/issues/232

**Pull request merged**